src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -722,6 +738,41 @@ private IConversionExpression CreateBoundAsOperatorOperation(BoundAsOperator bou
             return new LazyCSharpConversionExpression(operand, conversion, isExplicit, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit);
         }
 
+        private IDelegateCreationExpression CreateBoundDelegateCreationExpressionOperation(BoundDelegateCreationExpression boundDelegateCreationExpression)
+        {
+            Lazy<IOperation> target = new Lazy<IOperation>(() =>
+            {
+                if (boundDelegateCreationExpression.Argument is BoundMethodGroup boundMethodGroup &&
+                    boundMethodGroup.Methods.Length == 1)
```

Comment:
>boundMethodGroup.Methods.Length == 1 [](start = 20, length = 36)

I do not understand why do we need to have this restriction? What if there are several candidates?

Reply:
We have this restriction because we don't have support for unresolved BoundMethodGroups in V1. There's nowhere for me to expose the other methods that were considered.

Reply:
> We have this restriction because we don't have support for unresolved BoundMethodGroups in V1. There's nowhere for me to expose the other methods that were considered.

I am not sure I am following, couldn't there be a success case when there is more than one candidate. What happens then? Do we have a test for a scenario like that?

Reply:
I've updated this condition to check to see if `BoundDelegateCreationExpression.MethodOpt` is non-null. We're still not exposing other candidates if they were considered, but we'll only expose an actual method binding if a single method was selected at compile time. This won't cover a few valid scenarios (such as dynamic), but we need an actual `IMethodGroup` or something similar to solve that.